notebooks/usage.ipynb

```diff
@@ -0,0 +1,362 @@
+    "  - documentation\n",
+    "  - no support for pytorch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Code example"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "clone repo and install first from repo directory `python3 -m pip install .`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sys \n",
+    "sys.path.append(\"..\")\n",
+    "import iSparrow.sparrow_model_base as spm\n",
+    "import iSparrow.sparrow_recording as spr\n",
+    "import iSparrow.preprocessor_base as spb\n",
+    "import iSparrow.utils as utils\n",
+    "\n",
+    "import tests.make_mock_install as sp"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pathlib import Path\n",
+    "import pandas as pd"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# make a mock install of sparrow. will be invisible in the future\n",
+    "sp.install()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Demo\n",
+    "\n",
+    "- load model, preprocessor you want \n",
+    "- add recording to be analyzed and assign it the model, preprocessor to use \n",
+    "- analyze "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# variables for analysis\n",
+    "sigmoid_sensitivity = 1.0\n",
+    "num_threads = 12\n",
+    "min_conf = 0.25\n",
+    "\n",
+    "# variables for recording\n",
+    "recording_path = sp.EXAMPLES / \"soundscape.wav\"\n",
+    "sample_rate = 48000\n",
+    "overlap = 0.0\n",
+    "sample_secs = 3.0\n",
+    "resample_type = \"kaiser_fast\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ppd = utils.load_module(\"ppm\", sp.MODELS / Path(\"birdnet_default\") / \"preprocessor.py\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "md = utils.load_module(\"md\", sp.MODELS / Path(\"birdnet_default\") / \"model.py\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "preprocessor = ppd.Preprocessor(sample_rate=sample_rate, overlap=overlap, sample_secs=sample_secs, resample_type=resample_type)\n",
+    "\n",
+    "model = md.Model(sp.MODELS / \"birdnet_default\", num_threads=5, sigmoid_sensitivity=1.)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "recording = spr.SparrowRecording(preprocessor, model, sp.EXAMPLES / \"soundscape.wav\", min_conf=min_conf)"
```

Comment:
for example, use `recording_path` or better, rename to `recorded_sample` or something?